pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGeneratorTest.java

```diff
@@ -32,13 +32,22 @@
 import java.util.Random;
 import java.util.concurrent.ExecutionException;
 import java.util.zip.GZIPInputStream;
+import lombok.Cleanup;
 import org.apache.commons.io.IOUtils;
 import org.apache.pulsar.broker.PulsarService;
 import org.apache.pulsar.broker.ServiceConfiguration;
+import org.apache.pulsar.common.util.SimpleTextOutputStream;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 public class PrometheusMetricsGeneratorTest {
 
+
+    @BeforeClass
+    public static void setProperties() {
+        System.setProperty("io.netty.noUnsafe", "true");
```

Comment:
This could impact other tests running in the same JVM. A test that changes this property should be run independently. This will have no effect if Netty is already initialized when the test runs. I'd recommend creating a separate test class with the method if there's a need to test this in CI. 

I don't yet understand the motivation for supporting unsafe mode, so I also think that testing this is unecessary in the first place.

Reply:
@zymap Instead of setting properties, you can just mock a PlatformDependent to return true or false for unit testing.

Reply:
@codelipenghui Using Mockito causes other errors so I moved the test to another file. 

@lhotari The motication is we met some core dumps when running pulsar service. Then saw Netty suggested using io.netty.noUnsafe to see if it works. https://github.com/netty/netty/issues/2950. After adding that property, the core dump is gone, but the metrics failed to report because of the UnsupportedOpeartion exception. Then I made this support to use the direct buffer to avoid that exception.